src/server/server.js

```diff
@@ -197,6 +197,11 @@ Server.prototype._onSentryDown = function (sentryId) {
   var started = Date.now()
 
   nonblocking(presences).forEach(function (presence) {
+    // check if presence was destroyed
+    if (!presence || !presence.manager || !presence.manager.store) {
```

Comment:
this is super unclear. `ResourceManager#destroy()` and `Resource#destroy()` are synchronous, so `destroying` as an active state is not correct, and there is no reason to introduce a `destroyed` flag. What is the actual information about the system that is informing the business decision here?